src/Microsoft.DotNet.Build.Tasks/PackageFiles/executor/executor.cs

```diff
@@ -0,0 +1,227 @@
+            {
+                Executor executor = new Executor();
+                Setup jsonSetup = executor.openFile(args[0]);
+                if (jsonSetup != null)
+                {
+                    //will be passed by executor.cmd/sh? we only need to now if it is in windows or non-windows.
+                    string os = "windows";
+                    //default right now.
+                    string command = "simple";
+                    bool end = false;
+                    for (int i = 1; i < argsLength && !end; i++)
+                    {
+                        switch (args[i])
+                        {
+                            case "info":
+                                executor.usage();
+                                end = true;
+                                break;
+                            case "help":
+                                Console.WriteLine("Usage: build [command]:");
+                                ++i;
+                                if (i < argsLength)
+                                {
+                                    end = true;
+                                    if (args[i].Equals("properties"))
+                                    {
+                                        jsonSetup.ProvideHelpProperties();
+                                    }
+                                    else if (args[i].Equals("commands"))
+                                    {
+                                        jsonSetup.ProvideHelpInstructions(os);
+                                    }
+                                }
+                                if (!end)
+                                {
+                                    jsonSetup.help(os);
+                                    end = true;
+                                }
+                                break;
+                            default:
+                                command = args[i];
+                                break;
+                        }
+                    }
+                    
+                    if(!end)
+                    {
+                        if (jsonSetup.commands.ContainsKey(command))
+                        {
+                            Console.WriteLine(jsonSetup.buildCommand(jsonSetup.commands[command], os));
+                        }
+                        else
+                        {
+                            Console.WriteLine(string.Format("Error: Command {0} not found.", command));
+                            jsonSetup.help(os);
+                        }
+                    }
+                }
+                else
+                {
+                    Console.WriteLine("Error: Could not load Json configuration file.");
+                }
+            }
+            else
+            {
+                Console.WriteLine("Error: Json configuration file should be provided.");
+            }
+        }
+    }
+
+    public class Setup
```

Comment:
Perhaps have a base class and that instantiates an instance of a Windows help / build command class or Linux help /build command class / etc based on os, rather than passing an os string around and branching logic in various places.